packages/appkit/src/adapters/ChainAdapterBlueprint.ts

```diff
@@ -0,0 +1,426 @@
+
+  /**
+   * Signs a message with the connected wallet.
+   * @param {AdapterBlueprint.SignMessageParams} params - Parameters including message to sign, address, and optional provider
+   * @returns {Promise<AdapterBlueprint.SignMessageResult>} Object containing the signature
+   */
+  public abstract signMessage(
+    params: AdapterBlueprint.SignMessageParams
+  ): Promise<AdapterBlueprint.SignMessageResult>
+
+  /**
+   * Estimates gas for a transaction.
+   * @param {AdapterBlueprint.EstimateGasTransactionArgs} params - Parameters including address, to, data, and optional provider
+   * @returns {Promise<AdapterBlueprint.EstimateGasTransactionResult>} Object containing the gas estimate
+   */
+  public abstract estimateGas(
+    params: AdapterBlueprint.EstimateGasTransactionArgs
+  ): Promise<AdapterBlueprint.EstimateGasTransactionResult>
+
+  /**
+   * Sends a transaction.
+   * @param {AdapterBlueprint.SendTransactionParams} params - Parameters including address, to, data, value, gasPrice, gas, and optional provider
+   * @returns {Promise<AdapterBlueprint.SendTransactionResult>} Object containing the transaction hash
+   */
+  public abstract sendTransaction(
+    params: AdapterBlueprint.SendTransactionParams
+  ): Promise<AdapterBlueprint.SendTransactionResult>
+
+  /**
+   * Writes a contract transaction.
+   * @param {AdapterBlueprint.WriteContractParams} params - Parameters including receiver address, token amount, token address, from address, method, and ABI
+   * @returns {Promise<AdapterBlueprint.WriteContractResult>} Object containing the transaction hash
+   */
+  public abstract writeContract(
+    params: AdapterBlueprint.WriteContractParams
+  ): Promise<AdapterBlueprint.WriteContractResult>
+
+  /**
+   * Gets the ENS address for a given name.
+   * @param {AdapterBlueprint.GetEnsAddressParams} params - Parameters including name
+   * @returns {Promise<AdapterBlueprint.GetEnsAddressResult>} Object containing the ENS address
+   */
+  public abstract getEnsAddress(
+    params: AdapterBlueprint.GetEnsAddressParams
+  ): Promise<AdapterBlueprint.GetEnsAddressResult>
+
+  /**
+   * Parses a decimal string value into a bigint with the specified number of decimals.
+   * @param {AdapterBlueprint.ParseUnitsParams} params - Parameters including value and decimals
+   * @returns {AdapterBlueprint.ParseUnitsResult} The parsed bigint value
+   */
+  public abstract parseUnits(
+    params: AdapterBlueprint.ParseUnitsParams
+  ): AdapterBlueprint.ParseUnitsResult
+
+  /**
+   * Formats a bigint value into a decimal string with the specified number of decimals.
+   * @param {AdapterBlueprint.FormatUnitsParams} params - Parameters including value and decimals
+   * @returns {AdapterBlueprint.FormatUnitsResult} The formatted decimal string
+   */
+  public abstract formatUnits(
+    params: AdapterBlueprint.FormatUnitsParams
+  ): AdapterBlueprint.FormatUnitsResult
+}
+
+export namespace AdapterBlueprint {

```

Comment:
nit: we could move this to a different file

Reply:
As far as I know, to keep namespace typing and class with same name they must be in the same file.